### PR TITLE
Replace amp-story-bookend with amp-story-social-share

### LINKED ISSUE #6852: Replace amp-story-bookend with amp-story-social-share
## Task Description

As per https://github.com/ampproject/amphtml/issues/33218, we need to replace `<amp-story-bookend>` usage with the new `<amp-story-social-share>` component.

Mostly search+replace to change method name etc. in the `HTML` class, but the config is now also simpler.

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Change amp-story-bookend with amp-story-social-share. 

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

Requires you to [opting in to pre-release channels](https://github.com/ampproject/amphtml/blob/master/contributing/DEVELOPING.md?rgh-link-date=2021-03-19T17%3A42%3A03Z#opting-in-to-pre-release-channels) using the [experiments](https://cdn.ampproject.org/experiments.html) page. 

1. Open a story in the editor, click preview. 
2. Navigate to the end story. 
3. See share sheet.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6852

### CODE REVIEW

File: includes/Analytics.php [hunk 1]
```diff
@@ -189,28 +189,6 @@ public function get_default_configuration( $tracking_id ) {
 						'send_to'        => $tracking_id,
 					],
 				],
-				// Fired when the bookend is shown to the user, after the last page of the current story.
```
Comment (spacedmonkey) [inline includes/Analytics.php:192]: @swissspidy What should about these events?

Comment (swissspidy) [reply includes/Analytics.php:192]: Looks like those triggers are now superfluous and can be safely removed.

File: includes/Story_Renderer/HTML.php [hunk 2]
```diff
@@ -267,36 +267,22 @@ protected function print_analytics( $content ) {
...
-			],
-			[
-				'provider' => 'email',
-			],
-			[
-				'provider' => 'system',
-			],
-		];
+	protected function print_social_share( $content ) {
+		$share_providers = [ 'twitter', 'linkedin', 'email', 'system', ];
```
Comment (spacedmonkey) [inline includes/Story_Renderer/HTML.php]: Sadly the format of this array / object is different in the new component. Changing this format means it is not compatible with format of the old one and the filter. I think maybe we need to change the filter name here. Thoughts @swissspidy 

https://github.com/ampproject/amphtml/blob/d94427b5a9bc7ca1ef77718b2c4e25ed0469092e/extensions/amp-story/1.0/test/validator-amp-story-social-share.out#L34-L43

**Comment by swissspidy** [comment]
The `test_print_social_share` test now needs updating it seems.

Otherwise looks good!

**Comment by spacedmonkey** [comment]
> The test_print_social_share test now needs updating it seems.
That is what I get for reverting something at the end of the day before you leave. 

Reverted change to test now as well

**Comment by swissspidy** [comment]
This now seems to be in production in the AMP validator.

@spacedmonkey can you confirm that the pre-release opt-in is no longer necessary for this?

**Comment by spacedmonkey** [comment]
@swissspidy I am seeing on browsers with opting in beta channel now. 

Screenshot 2021-04-06 at 17 59 23

It should be safe to merge.

**Review by swissspidy — APPROVED** [review]

**Pull request merged**